linkml/generators/pydanticgen/array.py

```diff
@@ -175,6 +176,124 @@ def _validate(_v: Union[List[_T], list]):
 _ConListImports = Imports() + Import(module="pydantic", objects=[ObjectImport(name="conlist")])
 
 
+class ArrayValidator:
+    """
+    Validate the specification of a LinkML Array
+
+    .. todo::
+
+        It looks like :mod:`linkml.validator` is for validating instances against schema, rather
+        than validating the schema itself, so am not subclassing/writing as a plugin.
+        Unsure if there is a more general means of validating schema, but for now this is
+        an independent class
+    """
+
+    @classmethod
+    def validate(cls, array: ArrayExpression):
+        """
+        Validate an array expression.
+
+        Raises:
+            :class:`.ValidationError` if invalid
+        """
+        cls.array_exact_dimensions(array)
+        cls.array_consistent_n_dimensions(array)
+        cls.array_explicitly_unbounded(array)
+        cls.array_dimensions_ordinal(array)
+
+        if array.dimensions:
+            for dimension in array.dimensions:
+                cls.validate_dimension(dimension)
+
+    @classmethod
+    def validate_dimension(cls, dimension: DimensionExpression):
+        """
+        Validate a single array dimension
+
+        Raises:
+            :class:`.ValidationError` if invalid
+        """
+        cls.dimension_exact_cardinality(dimension)
+        cls.dimension_ordinal(dimension)
+
+    @staticmethod
+    def array_exact_dimensions(array: ArrayExpression):
+        """Arrays can have exact_number_dimensions OR min/max_number_dimensions, but not both"""
+        if array.exact_number_dimensions is not None and (
+            array.minimum_number_dimensions is not None or array.maximum_number_dimensions is not None
+        ):
+            raise ValidationError(
+                "Can only specify EITHER exact_number_dimensions OR minimum/maximum dimensions, " f"got: {array}"
```

Comment:
```suggestion
                "Can only specify EITHER exact_number_dimensions OR minimum/maximum dimensions, "
                f"got: {array}"
```

Reply:
huh weird it didn't commit in the batch and i think my comment got lost? i'll commit this but i think black will want it to be on one line, but if it does i'll combine the strings